portfolio/src/main/webapp/script.js

```diff
@@ -17,12 +17,80 @@
  */
 function addRandomFact() {
   const facts =
-      ['I have a pet dog', 'I like to code'];
+      ['I have a pet dog!', 'I like to code!', 'Marquees are my favourite HTML element!'];
 
   // Pick a random fact.
   const fact = facts[Math.floor(Math.random() * facts.length)];
 
   // Add it to the page.
   const factContainer = document.getElementById('fact-container');
   factContainer.innerText = fact;
-}
\ No newline at end of file
+}
+
+/**
+ * Controls slideshow image display on the page
+ */
+const deckNums = { // Enumerate slide decks
+  DOG_DECK: 0,
+  PLACE_DECK: 1,
+  FOOD_DECK: 2,
+};
+
+const PREV_SLIDE_DIR = -1;
+const NEXT_SLIDE_DIR = 1;
+
+class Slideshow {
+  constructor(idNum, slides, slideshowDOM) {
+    this.index = 0;
+    this.idNum = idNum;
+    this.slides = slides;
+    this.slideshowDOM = slideshowDOM;
+    this.prevButton = this.slideshowDOM.querySelectorAll(".prev-slide-button")[0];
+    this.nextButton = this.slideshowDOM.querySelectorAll(".next-slide-button")[0];
+    this.addButtons();
+    this.preloadImages();
+  }
+  changeSlide(direction) {
+    const deckSize = this.slides.length;
+    this.index += direction;
+    if(this.index === deckSize) { // wrap around
+      this.index = 0;
+    }
+    else if(this.index === -1) {
+      this.index = deckSize - 1;
+    }
+    console.log("clicked");
+    let imageDOM = this.slideshowDOM.querySelectorAll(".image")[0];
+    imageDOM.src = this.slides[this.index];
+  }
+  addButtons() {
+    let self = this;
+    this.prevButton.onclick = function() {
+      self.changeSlide(PREV_SLIDE_DIR);
+    };
+    this.nextButton.onclick = function() {
+      self.changeSlide(NEXT_SLIDE_DIR);
+    };
+  }
+  preloadImages() {
+    for(let i = 0; i < this.slides.length; i++) {
+      var img = new Image();
```

Comment:
I suggest using a for-of loop rather than iterating over indices. It makes your intent a bit clearer
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of

Reply:
For some reason, this prevented the preloading to work. My theory is that it makes a new variable for each slide and preloads that variable instead of the slides array variable, but I am not familiar how the Javascript-equivalent of pointers works in the context of arrays.

Reply:
Can you share the code that didn't work? I can't see why changing the loop format would break it.

Reply:
this is the edited function: 
`  preloadImages() { 
    for(const slide in this.slides) { 
      var img = new Image(); 
      img.src = slide; 
   }
}`
Do you want me to commit it for viewing?